src/server/game/Chat/Channels/Channel.cpp

```diff
@@ -262,13 +270,28 @@ void Channel::LeaveChannel(Player* player, bool send)
         // Update last_used timestamp in db
         UpdateChannelUseageInDB();
 
-        // If the channel owner left and there are still players inside, pick a new owner
+        // If the channel owner left and there are players still inside, pick a new owner
+        // do not pick invisible gm owner unless there only invisible gms in that channel (rare)
         if (changeowner && _ownershipEnabled && !_playersStore.empty())
         {
-            auto itr = _playersStore.begin();
-            ObjectGuid newowner = itr->first;
-            itr->second.SetModerator(true);
-            SetOwner(newowner);
+            bool found = false;
+            PlayerContainer::iterator itr;
```

Comment:
No, I need it to be mutable, see https://github.com/TrinityCore/TrinityCore/pull/17597/files/3f3018a1c4652a6a91b87f3a4799e3e047f9d724#diff-9b9481224e87b63170bb0bb1028e7a07R290